src/index.ts

```diff
@@ -0,0 +1,762 @@
+#!/usr/bin/env node
+import fs from 'node:fs'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+import spawn from 'cross-spawn'
+import mri from 'mri'
+import * as prompts from '@clack/prompts'
+import colors from 'picocolors'
+
+const {
+  blue,
+  blueBright,
+  cyan,
+  green,
+//   greenBright,
+  magenta,
+  red,
+  redBright,
+//   reset,
+  yellow,
+  magentaBright,
+} = colors
+
+const argv = mri<{
+  template?: string
+  help?: boolean
+  overwrite?: boolean
+}>(process.argv.slice(2), {
+  alias: { h: 'help', t: 'template' },
+  boolean: ['help', 'overwrite'],
+  string: ['template'],
+})
+const cwd = process.cwd()
+
+// prettier-ignore
+const helpMessage = `\
+Usage: create-vite [OPTION]... [DIRECTORY]
+
+Create a new Vite project in JavaScript or TypeScript.
+With no arguments, start the CLI in interactive mode.
+
+Options:
+  -t, --template NAME        use a specific template
+
+Available templates:
+${yellow    ('vanilla-ts     vanilla'  )}
+${green     ('vue-ts         vue'      )}
+${cyan      ('react-ts       react'    )}
+${cyan      ('react-swc-ts   react-swc')}
+${magenta   ('preact-ts      preact'   )}
+${redBright ('lit-ts         lit'      )}
+${red       ('svelte-ts      svelte'   )}
+${blue      ('solid-ts       solid'    )}
+${blueBright('qwik-ts        qwik'     )}`
```

Comment:
This needs to be updated. `create-vite` -> `create-vite-extra`. And the template names.